crates/nu-protocol/src/engine/capture_block.rs

```diff
@@ -1,9 +1,11 @@
+use std::collections::HashMap;
+
 use crate::{BlockId, Value, VarId};
 
 use serde::{Deserialize, Serialize};
 
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct Closure {
     pub block_id: BlockId,
-    pub captures: Vec<(VarId, Value)>,
+    pub captures: Box<HashMap<VarId, Value>>,
```

Comment:
I'm boxing it to reduce the memory layout of the struct. HashMap<VarId, Value> allocate 48 bytes while  Box<HashMap<VarId, Value>> only need 8 bytes.

Reply:
I'm not sure that reducing the size from 56 to 16 bytes justifies the extra indirection and potential overhead when accessing captures, especially since Nushell probably doesn't deal with a large number of Closure instances or pass them frequently between functions. With only a few dozen closures, it doesn't really make sense to optimize for passing them through registers instead of the program's call stack. This is even less relevant for something like `struct Stack`, which is so large that saving a few bytes is insignificant, and it will still be passed on the program's stack.

Reply:
I see your point, I will remove the box and update the benchmark later. Thanks for your opinions.